SpatialConvolution.lua

```diff
@@ -318,9 +318,9 @@ function SpatialConvolution:createIODescriptors(input)
         maxBufSize = math.max(maxBufSize, bufSize[1])
 
         self.extraBuffer = self.extraBuffer or cudnn.getSharedWorkspace()
-        self.extraBufferSizeInBytes = self.extraBuffer:nElement() * 4 -- float
+        self.extraBufferSizeInBytes = self.extraBuffer:nElement() * cudnn.sizeof(self.weight)
```

Comment:
Right, this and resize() should use self.extraBuffer.elementSize(). 
Which brings those questions: 
1) is TorchTensor always Float or does it depend on preferred type setting ?
2) is TorchTensor storage good for Double? Is it aligned properly?